### PR TITLE
STD placeholder stats: scan and report placeholder and disabled tests

### PR DESCRIPTION
**Pull request by rnetser**
## Summary

A CLI tool that scans the `tests/` directory for tests marked with `__test__ = False` and reports them in two categories:

### 1. Placeholder tests (not yet implemented)
Tests containing only a docstring describing expected behavior, with no actual implementation code.

### 2. Disabled tests (implemented but skipped)
Tests that have actual implementation (assertions, setup, etc.) but are marked with `__test__ = False`. These may need attention — the marker might have been left by mistake.

### Detection levels

- **Module-level**: `__test__ = False` at the top of the file — all tests in the module are affected
- **Class-level**: `__test__ = False` inside a class body — all methods in the class are affected
- **Method/function-level**: `test_method.__test__ = False` — only the specific test is affected

### Usage

```bash
uv run python std_placeholder_stats.py
uv run python std_placeholder_stats.py --tests-dir my_tests
uv run python std_placeholder_stats.py --output-format json
```

### Example text output

```
======== STD PLACEHOLDER TESTS (not yet implemented) ========

tests/network/test_ipv6.py::TestConnectivity
  - test_pod_to_pod
  - test_service_endpoint

Total: 2 placeholder tests in 1 file

======== DISABLED TESTS (implemented but marked __test__ = False) ========

tests/storage/test_snapshots.py::TestSnapshots
  - test_snapshot_restore

Total: 1 disabled test in 1 file
```

Co-authored-by: Claude <noreply@anthropic.com>

## Test plan

- [x] 50 unit tests across 14 test classes
- [x] Covers all detection levels (module, class, method, standalone function)
- [x] Placeholder vs disabled categorization verified
- [x] Both text and JSON output formats verified
- [x] Graceful handling of syntax errors, permission errors, and unreadable files

## Summary by CodeRabbit

* **New Features**
  * Added a utility tool to scan test directories and identify standard test design placeholder and disabled tests, with support for text and JSON output formats.

* **Tests**
  * Added comprehensive test suite covering scanning functionality, output formatting, and edge cases.

* **Chores**
  * Updated pre-commit configuration to exclude specific test directories from code quality checks.

### CODE REVIEW

File: scripts/std_placeholder_stats/tests/test_std_placeholder_stats.py [hunk 1]
```diff
@@ -0,0 +1,736 @@
...
+
+        file_paths = [pf.file_path for pf in result]
+        assert "tests/test_broken.py" not in file_paths, f"Unexpected 'tests/test_broken.py' in result: {file_paths}"
+        assert "tests/test_valid.py" in file_paths, f"Expected 'tests/test_valid.py' in result, got: {file_paths}"
+
+    def test_returns_empty_list_when_no_test_files(self, tests_dir: Path) -> None:
+        """scan_placeholder_tests() returns empty list when no test files exist."""
+        result = scan_placeholder_tests(tests_dir=tests_dir)
+
+        assert result == []
```
[the remark this answers is not in the document]
Comment (rnetser) [reply scripts/std_placeholder_stats/tests/test_std_placeholder_stats.py]: Fixed: added `test_handles_unreadable_files_gracefully` test

File: scripts/std_placeholder_stats/std_placeholder_stats.py [hunk 2]
```diff
@@ -0,0 +1,541 @@
...
+
+    Args:
+        placeholder_files: List of PlaceholderFile objects to count.
+
+    Returns:
+        A tuple of (total_tests, total_files).
+    """
+    total_tests = sum(pf.total_tests for pf in placeholder_files)
+    total_files = len(placeholder_files)
+    return total_tests, total_files
```
[the remark this answers is not in the document]
Comment (rnetser) [reply scripts/std_placeholder_stats/std_placeholder_stats.py]: Fixed: `count_placeholder_tests` now only counts files with `total_tests > 0`, excluding files that have only disabled tests.

**Comment by rnetser** [comment]
pre-commit.ci run
